f8a_worker/workers/git_stats.py

```diff
@@ -0,0 +1,159 @@
+import shutil
+
+import numpy as np
+from tempfile import mkdtemp
+from time import time
+from datetime import timedelta
+from sklearn.linear_model import LinearRegression
+
+from f8a_worker.process import Git
+from f8a_worker.base import BaseTask
+
+
+class GitStats(BaseTask):
+    """Collect and compute Git statistics."""
+
+    _SECONDS_PER_DAY = 60 * 60 * 24
+    _DAYS_PER_MONTH = 30
+    _DAYS_PER_YEAR = 365
+
+    @staticmethod
+    def _get_log(url):
+        """Clone Git repo and get its log.
+  
+        :param url: url to the git repo
+        """
+        dir = mkdtemp()
+
+        try:
+            git = Git.clone(url, dir)
+            # nice notebook to check at:
+            #   http://nbviewer.jupyter.org/github/tarmstrong/code-analysis/blob/master/IPythonReviewTime.ipynb
+            log = git.log()
+        finally:
+            shutil.rmtree(dir, ignore_errors=True)
+
+        return log
+
+    @staticmethod
+    def _get_average_changes(log):
+        """Compute average changes based on log.
+
+        :param log: log on which changes should be computed
+        :return: a tuple containing aggregated changes across the whole log
+        """
+        additions = 0
+        deletions = 0
+        count = 0
+        for entry in log:
+            for change in entry['changes']:
+                if isinstance(change[0], int):
+                    additions += change[0]
+                if isinstance(change[1], int):
+                    deletions += change[1]
+                count += 1
+
+        if count:
+            additions /= count
+            deletions /= count
+
+        return additions, deletions
+
+    @classmethod
+    def _get_trend(cls, log, starting_date):
+        """Get commit count trend based on log.
+
+        :param log: a log on which the trend should be computed
+        :param starting_date: starting date of log
+        :return: computed trend
+        """
+        records = [0]
+        date = starting_date
+        for entry in log:
+            if entry['author']['date'] > date + cls._SECONDS_PER_DAY:
+                date += cls._SECONDS_PER_DAY
+                records.append(0)
+            records[-1] += 1
+
+        lr = LinearRegression()
+        lr.fit(np.array(range(len(records))).reshape(-1, 1), np.array(records))
+
+        return lr.coef_[0]
+
+    @staticmethod
+    def _get_trend_str(trend):
+        """Convert numerical representation of trend to its string representation.
+
+        :param trend: numerical representation of thrend
+        :return: textual representation of trend
+        """
+        if trend > 3:
+            return 'increasing'
+        elif trend > -3:
+            return 'calm'
+        else:
+            return 'decreasing'
```

Comment:
Isn't the -3 to 3 boundary for 'calm' quite big ? I'd expect most of the trends to be between -3 to 3.